lib/src/migrators/module.dart

```diff
@@ -389,10 +398,23 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
   /// it exists, or as a global variable otherwise.
   void _declareVariable(VariableDeclaration node) {
     if (_localScope == null || node.isGlobal) {
-      if (node.isGuarded &&
-          _globalVariables.containsKey(node.name) &&
+      if (_globalVariables.containsKey(node.name) &&
           _globalVariables[node.name].span.sourceUrl != _currentUrl) {
-        _configuredVariables.add(_globalVariables[node.name]);
+        if (node.isGuarded) {
+          _configuredVariables.add(_globalVariables[node.name]);
+        } else if (!_upstreamStylesheets
+            .contains(_globalVariables[node.name].span.sourceUrl)) {
+          // This declaration reassigns a variable in another module. Since we
+          // don't care about the actual value of the variable while migrating,
+          // we leave the node in _globalVariables as-is, so that future
+          // references namespace based on the original declaration, not this
+          // reassignment.
```

Comment:
Nice comment that clearly explains why this code isn't doing something you might expect it to do :+1:.